pkg/capture/capture.go

```diff
@@ -504,6 +506,51 @@ func (c *Capture) process() {
 	}
 }
 
+func (c *Capture) rotate() (result rotateResult) {
```

Comment:
Refactored this so that everything that might incur a race is executed within the protective bounds of `rotationState`.